com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Math/Wave/NoiseSineWaveNode.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections.Generic;
+using static UnityEditor.ShaderGraph.Registry.Types.GraphType;
+
+namespace com.unity.shadergraph.defs
+{
+
+    internal class NoiseSineWaveNode : IStandardNode
+    {
+        public static FunctionDescriptor FunctionDescriptor => new(
+            1,     // Version
+            "NoiseSineWave", // Name
+            @"
+{
+    sinIn = sin(In);
+    Out = sinIn + lerp(MinMax.x, MinMax.y, frac(sin((sinIn - sin(In + 1.0)) * (12.9898 + 78.233))*43758.5453));
+}
+",
+            new ParameterDescriptor("In", TYPE.Vector, Usage.In),
+            new ParameterDescriptor("MinMax", TYPE.Vec2, Usage.In, new float[] { -0.5f, 0.5f }),
```

Comment:
Just calling this out because I think there's some design room here for MinMax usage of Vec2.
![image](https://user-images.githubusercontent.com/13984097/155633700-0e8890b2-cd8c-4a5e-9807-1db17d1c8d66.png)

Either separating them or making so the inline control is different-- maybe as a part of the UI Fields one of the options for Vectors is to make them use custom labels per component?

Reply:
I'm going to separate Min and Max into their own float inputs. Will this change cause headaches for translating from the old system to the new?
Safety = cost on the GPU. Because GPU performance is critical, we generally ask the user to manage that rather than forcing them to use our solution for it. That's my philosophy anyway. Feel free to override me if you think otherwise.